### PR TITLE
Allow to change WiFi channel width. Implements #10678

### PR DESCRIPTION
**Pull request by vktg**
- [X] Redmine Issue: https://redmine.pfsense.org/issues/10678
- [X] Ready for review

Allow to change 802.11n channel width (ht/20, ht/40, etc) from the WebGUI

### CODE REVIEW

File: src/etc/inc/interfaces.inc [hunk 1]
```diff
@@ -6799,6 +6804,72 @@ function get_wireless_modes($interface) {
...
 }
 
+function get_wireless_ht_modes($interface) {
+	/* return wireless HT */
+	$wireless_hts_supported = array(0 => gettext('Auto'));
+
+	$cloned_interface = get_real_interface($interface);
+
+	if ($cloned_interface && is_interface_wireless($cloned_interface)) {
+		$chan_list = "/sbin/ifconfig -v {$cloned_interface} list chan";
```
Comment (jim-p) [inline src/etc/inc/interfaces.inc:6814]: I know we have reasonable assurance that `$cloned_interface` is valid here but I would like to see that run through `escapeshellarg()` to be safe.

File: src/etc/inc/interfaces.inc [hunk 2]
```diff
@@ -6799,6 +6804,72 @@ function get_wireless_modes($interface) {
...
+		$stack_list = "/usr/bin/awk -F\"Channel \" '{ gsub(/\\*/, \" \"); print \$2 \"\\\n\" \$3 }'";
+		$format_list = "/usr/bin/awk '{print \$5 \",\" \$6 \",\" \$1}'";
+
+		$interface_channels = "";
+		exec("$chan_list | $stack_list | sort -u | $format_list 2>&1", $interface_channels);
```
Comment (jim-p) [inline src/etc/inc/interfaces.inc:6819]: `sort` needs a full path

File: src/etc/inc/interfaces.inc [hunk 3]
```diff
@@ -6799,6 +6804,72 @@ function get_wireless_modes($interface) {
...
```
Comment (jim-p) [inline src/etc/inc/interfaces.inc:6844]: Why repeat all of this? Make a common function and call it from both places.

Comment (vktg) [reply src/etc/inc/interfaces.inc]: all fixed

**Pull request merged**